lib/benchmark/benchmark.go

```diff
@@ -0,0 +1,112 @@
+package benchmark
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/HdrHistogram/hdrhistogram-go"
+	"github.com/gravitational/teleport/lib/client"
+	logrus "github.com/sirupsen/logrus"
+)
+
+const (
+	// PauseTimeBetweenBenchmarks is the time to pause between each benchmark
+	PauseTimeBetweenBenchmarks = time.Second * 5
+)
+
+// Generator provides a standardized way to get successive benchmarks from a consistent interface
+type Generator interface {
+	Generator() bool
+	GetBenchmark() (context.Context, Config, error)
+	Benchmark([]string, *client.TeleportClient) ([]*Result, error)
+}
+
+// Run is used to run the benchmarks, it is given a generator, command to run,
+// a host, host login, and proxy. If host login or proxy is an empty string, it will
+// use the default login
+func Run(cnf interface{}, cmd, host, login, proxy string) ([]*Result, error) {
+	var results []*Result
+	tc, err := makeTeleportClient(host, login, proxy)
+	if err != nil {
+		log.Fatalf("Unable to make teleport client: %v", err)
```

Comment:
let's never use `log.Fatalf` in library functions - it's causes the entire process to exit. Imagine someone is calling this as a library function, this will be unexpected and won't give caller any chance to handle this